azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -564,6 +583,49 @@ private void createPod(final int executionId) throws ExecutorManagerException {
     this.executorLoader.updateVersionSetId(executionId, versionSet.getVersionSetId());
   }
 
+  /**
+   * Items extracted from the podTemplate will be merged with the pod-spec. Merging criterion is
+   * such that, if the item in the already created pod-spec has the same name as of the item
+   * extracted from the template, then it will retained.
+   *
+   * @param podSpec     Already created podSpec using the {@link AzKubernetesV1SpecBuilder}
+   * @param podTemplate Instance of the class {@link AzKubernetesV1PodTemplate} to extract items
+   */
+  public static void mergePodSpec(V1PodSpec podSpec, AzKubernetesV1PodTemplate podTemplate) {
+    // Add those volumes which are not already available in the podSpec
+    List<V1Volume> podSpecVolumes = podSpec.getVolumes();
+    if (null != podSpecVolumes) {
```

Comment:
if podSpecVolumes != null and rest of the places in this PR

Reply:
Added the reasoning in the earlier comments.